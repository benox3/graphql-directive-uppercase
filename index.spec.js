const { graphql } = require('graphql');
const directive = require('./index.js');
const schema = require('./example/schema');

expect.extend({
  toBeUppercase(received) {
    if (typeof received !== 'string')
      return {
        message: () => `expected to be a string`,
        pass: false,
      };

    return {
      message: () => `expected ${received.toUpperCase()} got ${received}`,
      pass: received.toUpperCase() === received,
    };
  },
});

it('has defined initial declaration', () => {
  expect(directive.getDirectiveDeclaration()).toMatchSnapshot();
});

it('should change given String field to UPPER version', () => {
  const instance = new directive('upper');
  const field = {};

  instance.visitFieldDefinition(field);

  return expect(
    field.resolve(
      { personId: 5, name: 'vaclav brown' },
      {},
      {},
      { fieldName: 'name' }
    )
  ).resolves.toEqual('VACLAV BROWN');
});

it('should ignore any other type then String', () => {
  const instance = new directive('upper');
  const field = {};

  instance.visitFieldDefinition(field);

  return expect(
    field.resolve({ personId: 5, id: 0 }, {}, {}, { fieldName: 'id' })
  ).resolves.toEqual(0);
});

it('graphql query should return correctly modified entity', async () => {
  const response = await graphql(
    schema,
    `
      query persons {
        persons {
          name
        }
      }
    `
  );

  response.data.persons.forEach(({ name }) => {
    expect(name).toBeUppercase();
  });
});
