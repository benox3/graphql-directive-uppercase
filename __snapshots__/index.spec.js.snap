// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has defined initial declaration 1`] = `
GraphQLDirective {
  "args": Array [],
  "astNode": undefined,
  "description": "Converts (string) field to upper case form",
  "locations": Array [
    "FIELD_DEFINITION",
  ],
  "name": "upper",
}
`;
